### ISSUE TITLE
Not working with react-intl

### ISSUE THREAD

**Issue by mlarcher** [milankinen/livereactload#99]
When running with https://github.com/yahoo/react-intl (in latest version, 2.0.0-pr-3) we get an error related to the module exports.
In Chrome:
```
Uncaught TypeError: Cannot set property 'default' of undefined
```
in Firefox: 
```
TypeError: exports is undefined
```
this is happening in `node_modules/react-intl/node_modules/intl-relativeformat/index.js`
on the `exports['default'] = exports;` (last) line

**Comment by mlarcher** [comment]
Could you please tell me if this indeed is a bug in livereactload that you need time to fix our of I should open a ticket on the react-intl side?

**Comment by milankinen** [comment]
Hi @mlarcher and sorry about the delay!

Are you still getting this with the latest version of LiveReactload?

I tried to import `react-intl` to the basic example and tried live reloading with it - all went fine both in Chrome and FF.

**Comment by mlarcher** [comment]
Actually I was already on  the latest version (v2.1.1) when I got the error.
Are you using ES6 syntax for imports ?

**Comment by milankinen** [comment]
LiveReactload shouldn't mind about the import/export style at all. That's why it's strange that if fails with such error.

Do you have any example project that causes this error? I can't reproduce it myself. :-/

**Comment by mlarcher** [comment]
I'll try to narrow it down then, and get back to you with more info.

**Comment by mlarcher** [comment]
I managed to have livereactload work with my project and react-intl by using the same browserify setup as you do in the examples. It seems there is something going wrong when using the lrload plugin from gulp. I'll provide more info when I can, but maybe that would ring a bell or allow you to test differently ?

**Comment by mlarcher** [comment]
@milankinen I managed to strip everything down to a very simple version that still shows the problem. 
You can see it in action by just running `npm install` and then `npm start`. You will see that livereactload works properly at first, but will stop working with the reported error as soon as react-intl is uncommented in the src/project-root/app.jsx entry point.
[sample-app.zip](https://github.com/milankinen/livereactload/files/139914/sample-app.zip)

**Comment by milankinen** [comment]
Nice! Thank you very much! I'll test this today evening.

**Comment by milankinen** [comment]
Thanks @mlarcher for your fantastic ground work with this one! With your sample I was able to find the bug from LiveReactload. It seems that I was missing some logic when patching de-duped modules -- duplicate module definition didn't get `exports` from the original one which caused this error.

Many thanks to you! I'll release fixed version soon.

**Comment by mlarcher** [comment]
That's great new @milankinen ! Please let me know when the fix is released; I'm eager to try it out :)

**Comment by milankinen** [comment]
Released as a part of **[2.2.0](https://github.com/milankinen/livereactload/releases/tag/2.2.0)**

**Comment by mlarcher** [comment]
Thanks for the quick fix! I tested it successfully on my app :)
Unfortunately, I stumbled upon another issue with some other package, so I still can't make use of your hotreload feature :(
The issue is not related, so I'm closing this ticket and opening a new one at https://github.com/milankinen/livereactload/issues/104
